proxy.go

```diff
@@ -102,6 +102,21 @@ func NewProxy(config Config) *Proxy {
 	return proxy
 }
 
+// SetTimeout set Timeout of the proxy.
```

Comment:
This is just a read timeout, not a connection timeout, should be explained (or maybe we should rename both the property and the function?)